astronomer/providers/google/cloud/sensors/gcs.py

```diff
@@ -123,3 +124,39 @@ def execute_complete(
         if event["status"] == "success":
             return event["matches"]
         raise AirflowException(event["message"])
+
+
+class GCSObjectUpdateSensorAsync(GCSObjectUpdateSensor):
+    def __init__(
+        self,
+        poll_interval: float = 5,
+        **kwargs: Any,
+    ) -> None:
+        super().__init__(**kwargs)
+        self.poll_interval = poll_interval
+
+    def execute(self, context: "Context") -> None:
+        self.defer(
+            timeout=self.execution_timeout,
+            trigger=GCSCheckBlobUpdateTimeTrigger(
+                bucket=self.bucket,
+                object=self.object,
+                ts=context["data_interval_end"],
+                polling_period_seconds=self.polling_interval,
+                google_cloud_conn_id=self.google_cloud_conn_id,
+                hook_params=dict(delegate_to=self.delegate_to, impersonation_chain=self.impersonation_chain),
+            ),
+            method_name="execute_complete",
+        )
+
+    def execute_complete(
+        self, context: "Context", event: Optional[Dict[Any, Any]] = None
+    ) -> None:  # pylint: disable=unused-argument
```

Comment:
Shouldn't the return type be `Optional[str]` instead of `None` as per the implementation?